x/tokenomics/token_logic_module/validator_distribution_test.go

```diff
@@ -211,59 +165,330 @@ func TestDistributeValidatorRewards_ErrorCases(t *testing.T) {
 	}
 }
 
-func TestDistributeValidatorRewards_PrecisionHandling(t *testing.T) {
-	ctrl := gomock.NewController(t)
-	defer ctrl.Finish()
-
-	mockStakingKeeper := mocks.NewMockStakingKeeper(ctrl)
-
-	// Create validators with stakes that don't divide evenly
-	validators := []stakingtypes.Validator{
+// TestDistributeValidatorAndDelegatorRewards tests the combined validator and delegator reward distribution.
+// It verifies correct distribution to both validators and their delegators based on delegation amounts,
+// backward compatibility fallbacks, and precision handling for fractional distributions.
+func TestDistributeValidatorAndDelegatorRewards(t *testing.T) {
+	tests := []struct {
+		name                     string
+		validators               []stakingtypes.Validator
+		delegations              map[string][]stakingtypes.Delegation
+		rewardAmount             math.Int
+		expectedTransferCount    int
+		expectedValidatorRewards int64
+		expectedDelegatorRewards int64
+		opReason                 tokenomicstypes.SettlementOpReason
+		validation               func(*testing.T, *tokenomicstypes.ClaimSettlementResult)
+	}{
 		{
-			OperatorAddress: sample.ValOperatorAddressBech32(),
-			Tokens:          math.NewInt(333), // ~33.3% of total
-			Status:          stakingtypes.Bonded,
+			name: "success: mixed delegation amounts with equal self-bonded stakes",
+			validators: []stakingtypes.Validator{
+				createValidator(sample.ValOperatorAddressBech32(), 1_000_000),
+				createValidator(sample.ValOperatorAddressBech32(), 600_000),
+				createValidator(sample.ValOperatorAddressBech32(), 400_000),
+			},
+			rewardAmount:             math.NewInt(100_000),
+			expectedTransferCount:    5,      // 3 validators + 2 delegators
+			expectedValidatorRewards: 60_000, // 100k × 60% = 60,000 (self-bonded validator shares)
+			expectedDelegatorRewards: 40_000, // 100k × 40% = 40,000 (delegator shares)
+			opReason:                 tokenomicstypes.SettlementOpReason_TLM_GLOBAL_MINT_VALIDATOR_REWARD_DISTRIBUTION,
 		},
 		{
-			OperatorAddress: sample.ValOperatorAddressBech32(),
-			Tokens:          math.NewInt(333), // ~33.3% of total
-			Status:          stakingtypes.Bonded,
+			name: "success: equal self-bonded stakes receive equal rewards despite different total delegations",
```

Comment:
I see "900K, 600K, 300K". How is this equal self-bonded stakes?

Reply:
Comments added